### PR TITLE
Move some more cops to layout

### LINKED ISSUE #4704: Is EndAlignment not a Layout cop?
In #4278, a bunch of whitespace cops were moved into the new “Layout department”. Since, I have wondered why `Lint/EndAlignment` wasn’t moved along with them.

### PR DESCRIPTION
**Pull request by bquorning**
In #4278, many of the whitespace related cops were moved to the `Layout` namespace. But for various reasons, a few whitespace related cops remained in the `Lint` namespace.

Fixes #4704.

This PR moves  `Lint/EndAlignment`, `Lint/DefEndAlignment`, `Lint/BlockAlignment`, and `Lint/ConditionPosition` to the `Layout` namespace.

Things to consider:

* Ruby (`ruby -w`) will warn about the type of offenses found by `Lint/EndAlignment` and `Lint/DefEndAlignment`. Therefore, I have configured them with `Severity: warning` in default.yml.
* Ruby (again `ruby -w)` will *not* warn about the type of offenses found by `Lint/BlockAlignment` and `Lint/ConditionPosition`. Therefore, they now have the default `convention` severity level in RuboCop.
* `Lint/EndAlignment` and `Lint/DefEndAlignment` do have autocorrect implemented, but the feature was disabled in default.yml (see #1789). That choice is *unchanged* in this PR.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,10 @@
 * [#4477](https://github.com/bbatsov/rubocop/issues/4477): Warn when user configuration overrides other user configuration. ([@jonas054][])
 * [#5240](https://github.com/bbatsov/rubocop/pull/5240): Make `Style/StringHashKeys` to accepts environment variables. ([@pocke][])
 * [#5395](https://github.com/bbatsov/rubocop/pull/5395): Always exit 2 when specified configuration file does not exist. ([@pocke][])
+* [#4704](https://github.com/bbatsov/rubocop/issues/4704): Move `Lint/EndAlignment` to the `Layout` namespace. ([@bquorning][])
+* [#4704](https://github.com/bbatsov/rubocop/issues/4704): Move `Lint/EndAlignment` and `Lint/DefEndAlignment` to the `Layout` namespace. ([@bquorning][])
+* [#4704](https://github.com/bbatsov/rubocop/issues/4704): Move `Lint/EndAlignment`, `Lint/DefEndAlignment`, and `Lint/BlockAlignment` to the `Layout` namespace. ([@bquorning][])
+* [#4704](https://github.com/bbatsov/rubocop/issues/4704): Move `Lint/EndAlignment`, `Lint/DefEndAlignment`, `Lint/BlockAlignment`, and `Lint/ConditionPosition` to the `Layout` namespace. ([@bquorning][])
```
Comment (Drenmi) [inline CHANGELOG.md]: Seems like there's some duplication of entries here. 🙂

**Comment by jonas054** [comment]
:+1: The changes look good. This is an improvement over the current situation.

There are some intermittent failures on Travis CI. I restarted the build but the next one failed too.

File: CHANGELOG.md (hunk 1 above)
Comment (bquorning) [reply CHANGELOG.md]: Fixed. Sorry.

**Comment by bquorning** [comment]
I had to rebase to fix the mess I made in CHANGELOG…

**Comment by bbatsov** [comment]
> Ruby (again ruby -w) will not warn about the type of offenses found by Lint/BlockAlignment and Lint/ConditionPosition. Therefore, they now have the default convention severity level in RuboCop.

I really don't thin there's much difference between def end alignment and block end alignment. Maybe it's time to just dispense trying to mimic some questionable decisions made in MRI. 

I also see that that for the same of uniformity probably `BlockAlignment` should become `BlockEndAlignment`.

**Pull request merged**